projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMonitor.java

```diff
@@ -141,12 +141,15 @@ public void clearDataJobsGaugesNotIn(final Iterable<DataJob> dataJobs) {
     }
 
     /**
-     * Creates a gauge that exposes termination status for the specified data job.
-     * If a gauge already exists for the data job, it is updated if necessary.
+     * Creates metrics (a gauge and counters) that expose termination status
+     * for the specified data job and count the number of instances of the said
+     * status.
+     * If a gauge or counters already exist for the data job, they are updated
+     * if necessary.
      *
      * @param dataJob The data job for which to create or update a gauge.
      */
-    void updateDataJobTerminationStatusGauge(final DataJob dataJob) {
+    void updateDataJobTerminationStatusMetrics(final DataJob dataJob) {
```

Comment:
Can this function be called more than once. It did not matter when we were updating a gauge. But now it does. Since we can increment counter more than once by mistake.

Reply:
Logically speaking, yes it can be called more than once. I'm trying to figure out a way to test it, so I can implement some safeguard. Without a test, anything implemented would be as good as a bug.

For your first question:
> How are we going to ensure that different pods are showing the same number (assuming it's a problem)?
> And if they are showing the different number, would that be a problem and what kind of problem?

It would be a reporting problem, as when queried, it would skew reality showing a different number of executions and different statuses than expected.
Again, I need to figure out a way to test it.

> What happens if a pod is restarted? Its counter would be reset but the other pod would not be so Prometheus would scrape what?

I am not sure I understand completely the question. In general the rules for the termination gauges would apply to the counters, as well. If a pod is restarted, new termination gauge is created, as well as a new counter. If I understand correctly (though I might as well be wrong in my understanding) how the metrics work, they are time-based, so Prometheus would scrape the counters for the restarted pods with a different timestamp.